src/Eagle.ts

```diff
@@ -2475,13 +2490,71 @@ export class Eagle {
             }
         }
     }
+    
+    //dragdrop WIP
+
+    nodeDragStart = (eagle : Eagle, e : JQueryEventObject) => {
+        Eagle.nodeDropped = e.target;
+        $(".leftWindow").addClass("noDropTarget");
+        $(".rightWindow").addClass("noDropTarget");
+        $(".navbar").addClass("noDropTarget");
+
+        var drag = Eagle.nodeDropped.getElementsByClassName('input-group-prepend')[0] as HTMLElement;
+        (<DragEvent> e.originalEvent).dataTransfer.setDragImage(drag, 0, 0);
+        return true;
+    }
+
+
+    nodeDragEnd = (e : JQueryEventObject) => {
+        $(".leftWindow").removeClass("noDropTarget");
+        $(".rightWindow").removeClass("noDropTarget");
+        $(".navbar").removeClass("noDropTarget");
+        return true;    
+    }
+
+    nodeDragOver = (e : JQueryEventObject) => {
+        return false;    
+    }
+
+    nodeDrop = (eagle : Eagle,e : JQueryEventObject) => {
+        let nodeButton = Eagle.nodeDropped.getElementsByTagName('button')[0] as HTMLElement;
```

Comment:
Maybe just a slight line re-ordering here. Lines 2520 and 2523 seem related and should be together. I think 2522 is un-related and need not be in-between them. Could remove the commented code here too.